integration/logging_test.go

```diff
@@ -0,0 +1,147 @@
+		var err error
+		name, err = occam.RandomName()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	it.After(func() {
+		Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+		Expect(docker.Image.Remove.Execute(secondImage.ID)).To(Succeed())
+		Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+		Expect(os.RemoveAll(source)).To(Succeed())
+	})
+
+	context("when app is vendored", func() {
+		it("has correct logging output", func() {
+			var err error
+			source, err = occam.Source(filepath.Join("testdata", "vendored_app"))
+			Expect(err).NotTo(HaveOccurred())
+
+			var logs fmt.Stringer
+			image, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithBuildpacks(
+					minicondaBuildpack,
+					buildpack,
+					buildPlanBuildpack,
+				).
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), logs.String())
+
+			Expect(logs).To(ContainLines(
+				MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, buildpackInfo.Buildpack.Name)),
+				"  Executing build process",
+				fmt.Sprintf("    Running conda create --file /workspace/package-list.txt --prefix /layers/%s/conda-env --yes --quiet --channel /workspace/vendor --override-channels --offline",
+					strings.ReplaceAll(buildpackInfo.Buildpack.ID, "/", "_")),
+				MatchRegexp(`      Completed in (\d+m)?(\d+)(\.\d+)?(ms|s)`),
+				"",
+			))
+		})
+	})
+
+	context("when app have lockfile", func() {
+		it("has correct logging output", func() {
+			var err error
+			source, err = occam.Source(filepath.Join("testdata", "with_lock_file"))
+			Expect(err).NotTo(HaveOccurred())
+
+			var logs fmt.Stringer
+			image, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithBuildpacks(
+					minicondaBuildpack,
+					buildpack,
+					buildPlanBuildpack,
+				).
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), logs.String())
+
+			Expect(logs).To(ContainLines(
+				MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, buildpackInfo.Buildpack.Name)),
+				"  Executing build process",
+				fmt.Sprintf("    Running CONDA_PKGS_DIRS=/layers/%s/conda-env-cache conda create --file /workspace/package-list.txt --prefix /layers/%s/conda-env --yes --quiet",
+					strings.ReplaceAll(buildpackInfo.Buildpack.ID, "/", "_"), strings.ReplaceAll(buildpackInfo.Buildpack.ID, "/", "_")),
+				"    Running conda clean --packages --tarballs",
+				MatchRegexp(`      Completed in (\d+m)?(\d+)(\.\d+)?(ms|s)`),
+				"",
+			))
+			secondImage, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithBuildpacks(
+					minicondaBuildpack,
+					buildpack,
+					buildPlanBuildpack,
+				).
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), logs.String())
+
+			Expect(logs).To(ContainLines(
+				MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, buildpackInfo.Buildpack.Name)),
+				fmt.Sprintf("  Reusing cached layer /layers/%s/conda-env", strings.ReplaceAll(buildpackInfo.Buildpack.ID, "/", "_")),
+				"",
+			))
```

Comment:
Why is this test in this file? I feel like a test statement like this should be in the layer_reuse tests.

Reply:
We wanted to keep the logging assertions all in one file, to make this a true "logging" test and make the other tests completely logging-independent.

Reply:
I don't think that follows the pattern of many of our other buildpacks though. We will often have the logging expectations inside of the layer_reuse test. I don't think that this bothers me that much. You will need to fix the image removal issue by using a system similar to the layer_reuse tests

Reply:
Yeah, tbh I find it confusing that in other buildpacks we have a "logging test" but that logging assertions spill into other tests.
I'll address the image thing though.